### PR TITLE
[WIP] Generate static lib instead of reference lib

### LINKED ISSUE #18: libmatio.lib import library
looking at:

lib /list C:\conda\Library\lib\libmatio.lib

it seems that the libmatio.lib shipped is not a windows static library but a so-called import library [1]

it means that when linking against C:\conda\Library\lib\libmatio.lib the app produced
needs C:\conda\Library\bin\libmatio.dll

this makes it difficult to ship standalone applications, unless we hack our package system to copy the dll manually.

do you see an alternative? could the .lib on this conda build corresponds to a real static lib? note that on linux and mac we get both an archive .a and a shared lib (.so or .dylib)

[1] https://stackoverflow.com/questions/6402586/know-if-lib-is-static-or-import

cc @massich

### PR DESCRIPTION
**Pull request by massich**
Fix #18

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,6 +40,7 @@ test:
     - if not exist %PREFIX%\Library\bin\libmatio.dll exit 1  # [win]
     - if not exist %PREFIX%\Library\lib\libmatio.lib exit 1  # [win]
     - if not exist %PREFIX%\Library\include\matio.h exit 1  # [win]
+    - set xx=lib /list libmatio.lib # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: The linter is requesting another space before `# [win]`. Would get rid of any trailing whitespace as well.

**Comment by jakirkham** [comment]
Needs a build number bump.

**Comment by jakirkham** [comment]
Thoughts @tbeu?

**Comment by 183amir** [comment]
You need to bump the build number as well.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ source:
   md5: 2968b5a95f05fab8b79ec87f9a6c0cac
 
 build:
-  number: 0
+  number: 1
```
Comment (183amir) [inline recipe/meta.yaml:12]: please rebase on master and change build number to 2.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -40,6 +40,7 @@ test:
     - if not exist %PREFIX%\Library\bin\libmatio.dll exit 1  # [win]
     - if not exist %PREFIX%\Library\lib\libmatio.lib exit 1  # [win]
     - if not exist %PREFIX%\Library\include\matio.h exit 1  # [win]
+    - set xx=lib /list libmatio.lib  # [win]
```
Comment (183amir) [inline recipe/meta.yaml:43]: What does this line do?

File: recipe/bld.bat [hunk 4]
```diff
@@ -34,3 +34,6 @@ copy %SRC_DIR%\src\matio.h %LIBRARY_INC%\
 if errorlevel 1 exit 1
 copy %SRC_DIR%\visual_studio\matio_pubconf.h %LIBRARY_INC%\
 if errorlevel 1 exit 1
+
+:: XX
+lib /list %LIBRARY_LIB%\libmatio.lib
```
Comment (183amir) [inline recipe/bld.bat:39]: What does this line do?

**Review by 183amir — CHANGES_REQUESTED** [review]
Is this a WIP? or is it finished?

**Comment by traversaro** [comment]
Given CFEP-18 (https://github.com/conda-forge/cfep/blob/master/cfep-18.md), if someone wants to also package a static variant of `libmatio` it should do so by providing an additional `libmatio-static` output, instead of replacing the existing shared build. If anyone is interested in this, feel free to  open a PR in that direction, thanks!

**Pull request closed without merging**